Automattic/wp-calypso
# Tracks Opt-Out For Logged Out Users

We are improving the Tracks opt-out mechanism by making sure we do not record any Tracks events if the Analytics option has been disabled in the cookie banner.

Note the opt-out mechanism is already working for logged-in users.

Related to #79719 where we count the number of users opting out of analytics.

## Proposed Changes

* Within the function to record an event
    * Check for the `cookies sensitive_pixel_options`
    * If the Analytics option is turned off, we void recording the event

Something important here is that I am copy-pasting code from `client/lib/analytics/utils/` to `packages/calypso-analytics/src/utils/`. I would love to find a way to avoid the code duplication. Can we import files from there directly?
* `get-tracking-prefs.ts`
* `is-country-in-gdpr-zone.ts`
* `is-region-in-ccpa-zone.ts`


## Testing Instructions

* Run locally
* Make sure you see the cookie banner
* Unselect the Analytics option
* Confirm there are no more Tracks events with Tracks vigilante or watching network traffic

## Pre-merge Checklist
